internal/bminventory/inventory.go

```diff
@@ -3042,6 +3042,67 @@ func (b *bareMetalInventory) ResetCluster(ctx context.Context, params installer.
 	return installer.NewResetClusterAccepted().WithPayload(&c.Cluster)
 }
 
+func (b *bareMetalInventory) InstallHost(ctx context.Context, params installer.InstallHostParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	var h *models.Host
+	var cluster common.Cluster
+
+	log.Info("Install single day2 host: ", params.HostID)
+	err := b.db.Preload("Hosts").First(&cluster, "id = ?", params.ClusterID).Error
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	for i := range cluster.Hosts {
+		if *cluster.Hosts[i].ID == params.HostID {
+			h = cluster.Hosts[i]
+			break
+		}
+	}
+	if h == nil {
+		log.WithError(err).Errorf("host %s not found", params.HostID)
+		return common.NewApiError(http.StatusNotFound, err)
+	}
+
+	if !host.IsDay2Host(h) {
+		log.Errorf("InstallHost for host %s is forbidden: not a Day2 hosts", params.HostID.String())
+		return common.NewApiError(http.StatusConflict, fmt.Errorf("Method only allowed when adding hosts to an existing cluster"))
+	}
+
+	if swag.StringValue(h.Status) != models.HostStatusKnown {
+		log.Errorf("Install host for host %s, state %s is forbidden: host not in Known state", params.HostID.String(), swag.StringValue(h.Status))
+		return common.NewApiError(http.StatusConflict, fmt.Errorf("Cannot install host in state %s", swag.StringValue(h.Status)))
+	}
+
+	err = b.hostApi.AutoAssignRole(ctx, h, b.db)
+	if err != nil {
+		log.Errorf("Failed to update role for host %s", params.HostID)
+		return common.GenerateErrorResponder(err)
+	}
+
+	err = b.hostApi.RefreshStatus(ctx, h, b.db)
+	if err != nil {
+		log.Errorf("Failed to refresh host %s", params.HostID)
+		return common.GenerateErrorResponder(err)
+	}
+
+	if swag.StringValue(h.Status) != models.HostStatusKnown {
+		return common.NewApiError(http.StatusConflict, fmt.Errorf("Cannot install host in state %s after refresh", swag.StringValue(h.Status)))
+	}
```

Comment:
All this section doesn't belong here, it should part of host-api, `PrepareForInstallation`/`IsReadyForInstallation` etc.